### DISCUSSION TITLE
Using specific relay while having multiple reservation

### DISCUSSION THREAD

**Discussion by dariusc93** [libp2p/rust-libp2p#3776] [General]
Currently, while using the relay v2 protocol, it seems like all one would need to do is connect to a relay, and listen on the circuit relay and one can then talk to other peers over the relay, (assuming dcutr isnt used or a direct connection couldnt be establish and fallback), however say if you have reservation with 3 random nodes, 2 with default data limit (which I believe is about 256kb) and one with a higher data limit (eg 2mb), what would be the best course to insure that the relay with the highest data limit has higher priority over the others without closing the connection to the other two relay, or just only have a single reservation to the one with the high data limit?

**Comment by thomaseizinger** [comment]
At the moment, I am afraid we don't have any prioritization of which connection to prefer nor can the handlers of a connection influence, what the other handlers are doing.

In theory, it would be possible to add an API for that, similar to what we did with connection management: Have handlers block the creation of streams upon certain logic.

You could dynamically enable/disable behaviours based on how many direct connections you've established to the network.

@mxinden Any more ideas?

**Reply by mxinden** [reply]
I can only think of hacky solutions to the above problem.

Note that a relay informs both ends of the restrictions of a given relayed connection. See https://github.com/libp2p/specs/blob/master/relay/circuit-v2.md#stop-protocol. We could expose this information to the user, thus allowing a user to make more informed decisions either by disabling certain protocols per connection, or by restricting usage of a connection in their `ConnectionHandler` implementations itself.
